### ISSUE TITLE
Release candidate FreshRSS 1.19.2

### ISSUE THREAD

**Issue by Alkarex** [marienfressinaud/FreshRSS#4179]
Hello all,
Here is FreshRSS 1.19.2 [for final testing](https://github.com/FreshRSS/FreshRSS/milestone/52) before release.

It will probably be the last of the 1.19.x series (several new features are already done and waiting for the next 1.20 version).

Mainly some fixes in this minor release:
* Improve dropdown menus on mobile view [#4141](https://github.com/FreshRSS/FreshRSS/pull/4141), [#4128](https://github.com/FreshRSS/FreshRSS/pull/4128)
* Fix regression regarding keeping read state after seeing favourites / labels [#4178](https://github.com/FreshRSS/FreshRSS/pull/4178)
* Lots of code improvements, including improved support of PHP 8.1
* And more!

So please go ahead and upgrade your [git /edge branch](https://github.com/FreshRSS/FreshRSS), or [Docker :edge image](https://hub.docker.com/repository/docker/freshrss/freshrss), or [download an archive from edge](https://github.com/FreshRSS/FreshRSS/archive/edge.zip), and report any issue or comment, and also when it *just works* 😃

11 contributors for this release, including a few newcomers @marapavelka, @thomasrenes, @adamcstephens ... Thank you!

---

Detailed [tracked changes](https://github.com/FreshRSS/FreshRSS/compare/1.19.1...edge) (11 contributors, 35 commits, 384 changed files with 7,004 additions and 7,046 deletions.).

Full [changelog](https://github.com/FreshRSS/FreshRSS/blob/edge/CHANGELOG.md):

* Bug fixing
	* Fix regression regarding keeping read state after seeing favourites / labels [#4178](https://github.com/FreshRSS/FreshRSS/pull/4178)
	* Fix migration system on Synology and systems adding custom files to folders [#4163](https://github.com/FreshRSS/FreshRSS/pull/4163)
	* Fix wrong dropdown triangle UI for labels [#4174](https://github.com/FreshRSS/FreshRSS/pull/4174)
	* Fix minor UI bugs [#4169](https://github.com/FreshRSS/FreshRSS/pull/4169)
	* Fix minor SCSS details for the themes Ansum and Mapco [#4146](https://github.com/FreshRSS/FreshRSS/pull/4146)
* UI
	* Improve dropdown menus on mobile view [#4141](https://github.com/FreshRSS/FreshRSS/pull/4141), [#4128](https://github.com/FreshRSS/FreshRSS/pull/4128)
	* Improve menu icons [#4004](https://github.com/FreshRSS/FreshRSS/pull/4004)
* Deployment
	* Docker: development image `:newest` updated to PHP 8.1.1 and Apache 2.4.52 [#3666](https://github.com/FreshRSS/FreshRSS/pull/3666)
* i18n
	* Improve i18n CLI [#4131](https://github.com/FreshRSS/FreshRSS/pull/4131)
	* Use typographic quotes [#4133](https://github.com/FreshRSS/FreshRSS/pull/4133)
	* Improve message regarding forced feeds [#4145](https://github.com/FreshRSS/FreshRSS/pull/4145)
	* Improve Czech [#4151](https://github.com/FreshRSS/FreshRSS/pull/4151)
	* Improve English [#4161](https://github.com/FreshRSS/FreshRSS/pull/4161)
* Misc.
	* Increase PHPStan to [level 5](https://phpstan.org/user-guide/rule-levels) for code quality, also fixing several PHP 8.1 warnings [#4110](https://github.com/FreshRSS/FreshRSS/pull/4110), [#4123](https://github.com/FreshRSS/FreshRSS/pull/4123), [#4119](https://github.com/FreshRSS/FreshRSS/pull/4119)
	* Clean temporary files generated by automated tests [#4177](https://github.com/FreshRSS/FreshRSS/pull/4177)
	* Add automated spell checking of the code using [typos](https://github.com/crate-ci/typos) [#4138](https://github.com/FreshRSS/FreshRSS/pull/4138), [#4134](https://github.com/FreshRSS/FreshRSS/pull/4134)
	* Enforce code style *opening brace on same line* in PHPCS [#4122](https://github.com/FreshRSS/FreshRSS/pull/4122)
	* Remove broken GitHub Action automatically adding the `latest` tag to git [#4135](https://github.com/FreshRSS/FreshRSS/pull/4135)

**Comment by math-GH** [comment]
Looks fine and works for me as expected.

Did someone tested the installation routine?

**Comment by Alkarex** [comment]
No news, good news, let's ship! 🚀

**Comment by Bedeau69** [comment]
Bonjour,

La version 1.19.2 n'est pas proposée dans le système de mise à jour :

>  ⓘ Votre version actuelle de FreshRSS est la 1.19.1.
> Dernière vérification : 8 février 2022 à 08:37 

Est-ce normal ?
Merci

**Comment by marienfressinaud** [comment]
Oui il faut que je m'en occupe ! (dans la journée en principe)

**Comment by marienfressinaud** [comment]
Ça devrait être bon ! @Bedeau69

**Comment by aledeg** [comment]
@marienfressinaud Do you have a documentation of the process involved to add the new version on the server? Maybe we can automate that through github actions.

**Comment by Alkarex** [comment]
@aledeg While this is an option, I am more inclined to think that the update process could be rewritten to directly fetch the information from the source (the GitHub releases), avoiding the need of manual / automatic update of anything else.
I started that a long time ago but did not have the motivation to complete it. Help much welcome (a restart from scratch would also be fine) https://github.com/FreshRSS/FreshRSS/pull/1760

**Comment by marienfressinaud** [comment]
There is some documentation here: https://github.com/FreshRSS/FreshRSS/blob/edge/docs/en/developers/05_Release_new_version.md#updating-updatefreshrssorg

I didn't read it for a long time though. My current process to add a new version is quite simple (cf. https://github.com/FreshRSS/update.freshrss.org/commit/4b09f0d7f4b77135c4ee3c0e3a687f4ef362c9d7):

- I update the file `versions.php` (i.e. a line for the dev version + a line for the current version pointing to a non existing version + I update the last versions to point to the current version)
- I copy-paste the last script `scripts/update_to_x.y.z.php`
- I change the constant `PACKAGE_URL` (I don't touch to the rest of the script, it's mainly useless now, but the variable `$dirs_to_check` should probably be checked)
- on the server, I just have to do a `git pull`

It's really not a big deal to do this manually and it's the occasion to verify that the server runs correctly ^^

But I agree with @Alkarex, we can remove this intermediate server. I wanted to work on this in 2020 (that's why I added the migration system), but I totally lost my motivation. I took some notes on how to improve the current process that I never shared, I'll try to share them this weekend.

Edit: see https://github.com/FreshRSS/FreshRSS/discussions/4251
